modules/ort/src/main/kotlin/workflow/processors/enricher/OrtDownloaderProcessor.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.ort.workflow.processors.enricher
+
+import com.here.ort.downloader.DownloadException
+import com.here.ort.downloader.Downloader
+import com.here.ort.utils.encodeOrUnknown
+import com.here.ort.utils.packZip
+import org.eclipse.sw360.antenna.model.artifact.Artifact
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactSourceFile
+import org.slf4j.LoggerFactory
+import org.eclipse.sw360.antenna.ort.utils.ArtifactToPackageMapper
+import java.io.File
+import org.eclipse.sw360.antenna.api.workflow.AbstractProcessor
+import org.eclipse.sw360.antenna.model.artifact.ArtifactCoordinates
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactFilename
+
+class OrtDownloaderProcessor : AbstractProcessor() {
+    companion object {
+        const val ORT_DOWNLOADER_DIR = "ort-downloader-result"
+        private val LOGGER = LoggerFactory.getLogger(OrtDownloaderProcessor::class.java)
+    }
+
+    init {
+        workflowStepOrder = 350
+    }
+
+    private lateinit var ortDownloadFolder: File
+
+    override fun configure(configMap: MutableMap<String, String>) {
+        ortDownloadFolder = context.toolConfiguration.antennaTargetDirectory.resolve(ORT_DOWNLOADER_DIR).toFile()
+        ortDownloadFolder.takeUnless { it.exists() }?.let { it.mkdirs() }
+    }
+
+    override fun process(intermediates: MutableCollection<Artifact>): MutableCollection<Artifact> {
+        intermediates.filterNot {
+            it.sourceFile.isPresent
+        }.forEach {
+            val pkg = ArtifactToPackageMapper().apply(it)
+            val name = it.askFor(ArtifactCoordinates::class.java).takeIf { f -> f.isPresent }
+                    ?.let {
+                        it.get().mainCoordinate.canonicalize()
+                    } ?: it.askFor(ArtifactFilename::class.java).takeIf { f -> f.isPresent }
+                    ?.let { f -> f.get().bestFilenameEntryGuess.takeIf { fe -> fe.isPresent }
+                            ?.let { e -> e.get().filename }
+                    } ?: "unknown"
```

Comment:
Here's a slightly reformatted variant of the above which also has a few redundant `let`s removed (simply apply the IDE inspections):

            val name = artifact.askFor(ArtifactCoordinates::class.java).takeIf {
                it.isPresent
            }?.get()?.mainCoordinate?.canonicalize() ?: artifact.askFor(ArtifactFilename::class.java).takeIf {
                it.isPresent
            }?.let { filename ->
                filename.get().bestFilenameEntryGuess.takeIf {
                    it.isPresent
                }?.get()?.filename
            } ?: "unknown"